### DISCUSSION TITLE
Why the cryptographically secure random generator requirement?

### DISCUSSION THREAD

**Discussion by PlaceReporter99** [Vyxal/Vyxal#1770] [Q&A] [answered]
[In this document](https://github.com/Vyxal/Vyxal/blob/main/documents/specs/Transpilation.md#:~:text=If%20no%20loop%20variable%20is%20provided%2C%20a%20loop%20variable%20is%20to%20be%20generated%20using%20%22LOOP%22%20%2B%20secrets.token_hex(16)), it says that the loop variable must be generated using `"LOOP"+secrets.token_hex(16)` if no loop variable name is given. I don't get why you need the `secrets` module, as I see no reason this needs to be cryptographically secure since it's just a loop variable.

**Comment by ysthakur** [comment, marked as answer]
I'm assuming it was just used because it's a convenient way to generate random strings with a low risk of generating the same string twice. If you wanted to use the `random` module instead, you'd need something a lot longer, like maybe `''.join(random.choices(string.ascii_uppercase + string.digits, k=16))` (from [this answer](https://stackoverflow.com/a/2257449))
